src/oop/special_class_method/medium/q7/MainTest.java

```diff
@@ -0,0 +1,205 @@
+package oop.special_class_method.medium.q7;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String className = "PiggyBank";
+    private final String attributeName1 = "numPennies";
+    private final String getAttributeMethodName1 = "getNumPennies";
+    private final String setAttributeMethodName1 = "setNumPennies";
+    private final String attributeName2 = "numNickels";
+    private final String getAttributeMethodName2 = "getNumNickels";
+    private final String setAttributeMethodName2 = "setNumNickels";
+    private final String toStringMessage = "The total amount is $";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q7." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided portions of class " + className + ". Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName1, int.class), modifiedClassMessage);
+        assertTrue(testClass.hasModifier(attributeName1, "private"), modifiedClassMessage);
+        assertTrue(testClass.hasField(attributeName2, int.class), modifiedClassMessage);
+        assertTrue(testClass.hasModifier(attributeName2, "private"), modifiedClassMessage);
+        Class<?>[] classes = {};
+        Object testInstance = testClass.createInstance();
+        _assertEquals(0, testClass.getFieldValue(testInstance, attributeName1), modifiedClassMessage);
+        _assertEquals(0, testClass.getFieldValue(testInstance, attributeName2), modifiedClassMessage);
+        assertTrue(testClass.hasModifier(classes, "public"), modifiedClassMessage);
+    }
+
+
+    private static Stream<Integer> intInputProvider() {
+        return Stream.of(
+                0, 23, 553233333, 111111, 3554
+        );
+    }
+
```

Comment:
I feel like we should be testing more than one case for the get methods

Reply:
Since the constructor only assigns them to 0, we can't test them without using the setters, which will also test the validity of setters and we don't want that. We already have more than one test in worksTogether for the getter.

Reply:
Never mind I can just use the setField.